src/pexprs-toArgumentNameList.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+// --------------------------------------------------------------------
+// Imports
+// --------------------------------------------------------------------
+
+var common = require('./common');
+var pexprs = require('./pexprs');
+
+// --------------------------------------------------------------------
+// Operations
+// --------------------------------------------------------------------
+
+/*
+  The `toArgumentNameList` method returns a list of strings that will
+  be used as the default argument names for its receiver (a pexpr) in a
+  semantic action. This is used exclusively by the Semantics Editor.
+*/
+
+pexprs.PExpr.prototype.toArgumentNameList = common.abstract;
+
+pexprs.Prim.prototype.toArgumentNameList = function(optIndex) {
+  if (/^[a-zA-Z0-9]+$/.test(this.obj)) {
+    return ['_' + this.obj];
+  } else {
+    return ['$' + optIndex];
+  }
+};
+
+pexprs.Range.prototype.toArgumentNameList = function(optIndex) {
+  return [this.from + '_to_' + this.to];
+};
+
+pexprs.Alt.prototype.toArgumentNameList = function(optIndex) {
+  var termArgumentNames = this.terms.map(function(term, idx) {
+    return term.toArgumentNameList(idx + 1);
+  });
+
+  var ans = [];
+  termArgumentNames.forEach(function(termArgumentName) {
+    termArgumentName.forEach(function(termSubArgumentName, idx) {
+      if (idx >= ans.length) {
+        ans[idx] = [termSubArgumentName];
+      } else if (ans[idx].indexOf(termSubArgumentName) < 0) {
+        ans[idx].push(termSubArgumentName);
+      }
+    });
+  });
+
+  return ans.map(function(argumentList) {
+    return argumentList.join('_or_');
+  });
+};
+
+pexprs.Seq.prototype.toArgumentNameList = function(optIndex) {
+  var ans = [];
+  var count = {};
```

Comment:
Nit: you should change `{}` to `Object.create(null)`, or else names like `toString` won't do the right thing.